feat(python): Add Series.degrees/radians and clarify trig functions use radians

mdavis-xyz opened the pull request:
Fixes #22950

I have been unable to run `make pre-commit`, `make fmt`, `make doctest`, `make html`, because I get errors related to `make`.

```
$ make pre-commit
.venv/bin/ruff check
bash: line 1: .venv/bin/ruff: No such file or directory
make: *** [Makefile:135: fmt] Error 127
```

So I haven't actually tested that these changes render properly as a web page.

These changes are only for python. I have not yet found where the corresponding doc strings are for Rust.

mdavis-xyz committed:
docs: Mention trig functions use radians not deg (#22950)

mdavis-xyz commented:
I just spent a few hours trying, and failing, to install the dependencies required to build polars, run unit tests, linting etc. Even on the main branch, I get errors with `make doctest` (for stuff I haven't touched).
The [setup instructions](https://docs.pola.rs/development/contributing/#installing-dependencies) don't work. I keep getting errors about maturin not existing. I tried this multiple times on my laptop, and even a fresh VM. Eventually I got it working. I think the issue is that the initial build can fail at a point which is unrecoverable. i.e. `make clean` won't work because the initial make didn't finish. But `make test` won't work because it didn't finish creating dependencies (e.g. creating `.venv`.)

So I've just pushed what I've got. Hopefully the CICD can tell us if it's good. Then someone with a working dev environment can maybe run some tests.

mdavis-xyz committed:
Merge branch 'main' into deg_rad_doc_22950

mdavis-xyz commented:
Solves #18783

mdavis-xyz committed:
Merge branch 'main' into deg_rad_doc_22950

mdavis-xyz committed:
Merge branch 'main' into deg_rad_doc_22950

mdavis-xyz committed:
Merge branch 'main' into deg_rad_doc_22950

DeflateAwning commented:
@orlp Would you mind reviewing this PR? Super simple, but a very important detail to include in the docs.

mdavis-xyz committed:
Merge branch 'main' into deg_rad_doc_22950

mdavis-xyz committed:
Merge branch 'main' into deg_rad_doc_22950

mdavis-xyz commented:
Can someone please review this PR?

It's ready to merge, and I have other PRs ready to merge which can't be marked as ready because I've hit the 2 PR limit.

TNieuwdorp reviewed (changes requested):
Hi @mdavis-xyz, thanks for the PR! I verified all the new doctest examples and they all match exactly, so nice work getting those right without being able to run the suite. A few changes needed before we can merge:

### Move the Notes sections above Examples

Everywhere else in the codebase (and per numpydoc convention) Notes comes before Examples. Could you reorder that in all the touched docstrings?

### Minor cleanups

- [ ] Only `Expr.arcsin`'s Returns section got the "...measuring an angle in radians" wording. Could you add it to arccos/arctan too?
- [ ] On the Series methods, "The argument must be in radians" reads a bit oddly since they take no argument. Maybe "Input values are interpreted as radians."?
- [ ] The Rust docstrings (under crates/polars-plan/src/dsl/) can be a follow-up PR.


@orlp This PR adds a new public API, is that okay, or should it go into a separate PR?

If so, there are some follow-up tasks:
- [ ] Add Series.degrees and Series.radians to py-polars/docs/source/reference/series/computation.rst so they show up in the API reference.
- [ ] Mention the new methods explicitly in the PR description.

---

Re: your make errors: `.venv/bin/ruff: No such file or directory` just means the virtualenv hasn't been set up yet. Run `make requirements` first (it creates .venv and installs the dev dependencies), then `make fmt` and `make doctest` should work. You can refer to

mdavis-xyz commented:
Thanks! Ah, there are more issues than I expected. I'll resolve those in the next week.

I didn't intend to add a new API. At least, I don't recall so. (I wrote this so long ago I can't quite remember.) I suspect the new `degrees()` function was an existing function that moved, and somehow when I rebased to the latest master, git didn't merge the way I wanted. I'll look into that.

orlp commented:
I think this doesn't *really* introduce a new API, but merely documents `degrees` for `Series`. That is fine.

We already have `Expr.degrees`, it's an accident a corresponding method doesn't exist on `Series`.

mdavis-xyz committed:
docs(python): Move notes before examples in trig functions

mdavis-xyz committed:
docs(python): Mention radians vs degrees in Expr arccos and arctan, like arcsin

mdavis-xyz committed:
docs(python): Rephrase radian explanation for series trig functions

mdavis-xyz committed:
python(docs): Expose existing Series degrees and radians functions in docs

mdavis-xyz commented:
I've just implemented:

- Notes moved before examples
- Added `measuring an angle in radians` for `Expr` `arccos`, `arctan`. 
- Rephrased "The argument must be in radians"
- Add Series.degrees and Series.radians to py-polars/docs/source/reference/series/computation.rst

As for "Mention the new methods explicitly in the PR description.", I don't know where you want that. Should I edit the highest comment in this thread? Or are you referring to the PR title? I've just changed the title.
